instrumentation/android-instrumentation/src/main/java/io/opentelemetry/android/instrumentation/internal/AndroidInstrumentationLoaderImpl.kt

```diff
@@ -28,7 +28,7 @@ class AndroidInstrumentationLoaderImpl : AndroidInstrumentationLoader {
 
     @Throws(IllegalStateException::class)
     fun registerForTest(instrumentation: AndroidInstrumentation) {
-        if (instrumentation::class.java in instrumentations) {
+        check(instrumentation::class.java !in instrumentations) {
             throw IllegalStateException("Instrumentation with type '${instrumentation::class.java}' already exists.")
         }
```

Comment:
I think you don't need to throw, just return a string from the closure.

```suggestion
fun registerForTest(instrumentation: AndroidInstrumentation) {
    check(instrumentation::class.java !in instrumentations) {
        "Instrumentation with type '${instrumentation::class.java}' already exists."
    }
```

Reply:
Ugh, of course. I was relying on the tests passing, but apparently this wrong way of doing it also gets past detekt lol